core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -229,6 +288,68 @@ protected final void requestMessagesFromDeframer(int numMessages) {
     }
   }
 
+  /**
+   * Event handler to be called by the subclass when a number of bytes are being queued for sending
+   * to the remote endpoint.
+   *
+   * @param numBytes the number of bytes being sent.
+   */
+  protected final void onSendingBytes(int numBytes) {
+    synchronized (onReadyLock) {
+      shouldNotifyOnReady = true;
```

Comment:
If you are going to have a separate isReady() call, then we only want to call onReady() if isReady() could have returned false.

I'd suggest this:
```
numSentBytesQueued += numBytes;
if (numSentBytesQueued >= onReadyThreshold) {
  shouldNotifyOnReady = true;
}
```

You could also use:
```
numSentBytesQueued += numBytes;
if (!isReady()) {
  shouldNotifyOnReady = true;
}
```